coil-gif/src/nonAndroidMain/kotlin/coil3/gif/AnimatedSkiaImage.kt

```diff
@@ -0,0 +1,233 @@
+package coil3.gif
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.setValue
+import coil3.Canvas
+import coil3.Image
+import kotlin.time.TimeMark
+import kotlin.time.TimeSource
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import org.jetbrains.skia.AnimationFrameInfo
+import org.jetbrains.skia.Bitmap
+import org.jetbrains.skia.Codec
+
+internal class AnimatedSkiaImage(
+    private val codec: Codec,
+    private val coroutineScope: CoroutineScope,
+    private val timeSource: TimeSource,
```

Comment:
Great idea to pass a time source here. [Copied](https://github.com/coil-kt/coil/pull/2723) it `CrossfadePainter` as well.